app/models/event.rb

```diff
@@ -113,20 +121,32 @@ def average_rating
     @total_rating > 0 ? number_with_precision(@total_rating / @total.to_f, precision: 2, strip_insignificant_zeros: true) : 0
   end
 
-  def submitter
-    result = event_users.where(event_role: 'submitter').first
-    if result.nil?
-      user = nil
-      # Perhaps the event_users haven't been saved, if this is a new proposal
-      event_users.each do |u|
-        if u.event_role == 'submitter'
-          user = u.user
-        end
-      end
-      user
-    else
-      result.user
+  # def submitter
+  #   result = event_users.where(event_role: 'submitter').first
+  #   if result.nil?
+  #     user = nil
+  #     # Perhaps the event_users haven't been saved, if this is a new proposal
+  #     event_users.each do |u|
+  #       if u.event_role == 'submitter'
+  #         user = u.user
+  #       end
+  #     end
+  #     user
+  #   else
+  #     result.user
+  #   end
+  # end
+
+  # get event speakers with the event sumbmitter at the first position
+  # if the submitter is also a speaker for this event
+  def speakers_ordered
+    @speakers = speakers.to_a
+    @speakers << submitter unless @speakers.any?
+
+    if @speakers.reject! { |speaker| speaker == submitter }
+      @speakers.insert(0, submitter)
```

Comment:
Line 144 adds the submitter as a speaker if there's no speakers.
Then line 146 deletes the submitter if he's one of the speakers, and then line 147 adds him again. 😕 

I'm wondering: should an event be valid if it has no speakers? Then we wouldn't need line 144.

How about:

```ruby
def speakers_ordered
  speakers_list = speakers.to_a
  submitter_index = speakers_list.find_index(submitter)
  speakers_list.unshift(speakers_list.delete_at(submitter_index)) if submitter_index && submitter_index > 0
  speakers_list
end
```